src/eve/visitors.py

```diff
@@ -29,23 +29,67 @@
 class NodeVisitor:
     """Simple node visitor class based on :class:`ast.NodeVisitor`.
 
-    The base class walks the tree and calls a visitor function for every
-    node found. This function may return a value which is forwarded by
-    the `visit` method. This class is meant to be subclassed, with the
+    A NodeVisitor instance walks a node tree and calls a visitor
+    function for every item found. Return values from specific visitor
+    methods (not :meth:`generic_visit`) are forwarded by the
```

Comment:
The comment about return values is not 100% clear to me (after our discussino in the morngin).

Reply:
Improved docs